### PR TITLE
refactor: Create a specific submodule to store custom exceptions

### PR DESCRIPTION
**Pull request by glemaitre**
Store all custom exceptions in a specific submodule for the CLI.

### CODE REVIEW

**Comment by glemaitre** [comment]
If you have over exceptions in the library, I agree that the root folder makes more sense.

File: skore/src/skore/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+"""Custom exceptions for the CLI."""
```
Comment (thomass-dev) [inline skore/src/skore/exceptions.py:1]:
```suggestion
"""Custom exceptions."""
```

**Review by thomass-dev — CHANGES_REQUESTED** [review]
Thanks, its ready to merge for me except the suggested change. @augustebaum can you take a look?

**Review by thomass-dev — APPROVED** [review]

File: skore/src/skore/cli/launch_dashboard.py [hunk 2]
```diff
@@ -13,12 +13,6 @@
 from skore.ui.app import create_app
 
 
-class ProjectNotFound(Exception):
-    """Project was not found."""
-
-    project_path: Path
```
Comment (thomass-dev) [inline skore/src/skore/cli/launch_dashboard.py:19]: Note that this is not used anywhere.

**Pull request merged**